qiskit/utils/quantum_instance.py

```diff
@@ -124,15 +124,24 @@ def __init__(self,
         self._pass_manager = pass_manager
 
         # setup run config
-        if shots is not None:
-            if self.is_statevector and shots != 1:
-                logger.info("statevector backend only works with shot=1, changing "
-                            "shots from %s to 1.", shots)
-                shots = 1
-            max_shots = self._backend.configuration().max_shots
-            if max_shots is not None and shots > max_shots:
-                raise QiskitError('The maximum shots supported by the selected backend is {} '
-                                  'but you specified {}'.format(max_shots, shots))
+        if self.is_statevector and shots != 1:
+            logger.info("statevector backend only works with shot=1, changing "
+                        "shots from %s to 1.", shots)
+            shots = 1
+
+        # TODO: exchange this for a Backend instancecheck once Aer's simulators implement that
+        # interface
+        if hasattr(backend, 'options'):
+            if 'shots' in backend.options:
+                if shots != backend.options['shots']:
+                    logger.info('Overwriting the number of shots in the quantum instance with the '
+                                'settings from the backend.')
+                shots = backend.options.get('shots', 1024)
```

Comment:
Is there any reason why this is outdented. I get that at this point they are the same but then there is no reason to set it, so it could be nested under the if,  and the default value there seems like it would not be needed as it was tested against whatever value it contained not 1024.

Reply:
I tried running it with explicitly setting `shots` to `None` and that broke, so I assumed that this was not a needed check. Is there a scenario where you explicitly set `shots` to `None`?

Reply:
But I can revert to what it was before.

Reply:
I do not know why it was testing it to None before. Maybe that was a left over from earlier code. My comment was more that the updating of shots ie shots = backend.options.... could be done inside the if test above, along with the warning, so it was only set if they were not equal. Outside it ends up setting it always, which if its the same does not matter. Getting it with a default value of 1024 seems not needed since the presence of the key was already tested. It does no harm of course.